src/test/java/io/jenkins/infra/repository_permissions_updater/helper/URLHelper.java

```diff
@@ -0,0 +1,66 @@
+package io.jenkins.infra.repository_permissions_updater.helper;
+
+import java.net.URL;
+import java.net.URLStreamHandlerFactory;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class manage the handling of fake url for URL creation
+ */
+public sealed interface URLHelper permits URLHelper.URLHelperImpl {
```

Comment:
this class and a bunch of other mocks could be avoided by using https://wiremock.org/ instead of mocking the inputstream responses.

Wiremock would also mean less re-work on tests required if someone gets around to modernising the http code.

Not a blocker just a suggestion if you have time (I'll merge this now, but if you want to do a follow up that would be appreciated).

Reply:
I would rather do it as downstream PR